.github/workflows/ci.yml

```diff
@@ -2,9 +2,27 @@ name: CI
 
 on:
   pull_request:
+    branches:
+      - main
 
 jobs:
+  format-protobuf:
+    name: Format Protobuf
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+          submodules: true
+
+      - name: Check Diff
+        run: docker run --rm --volume "$(pwd)/Protos/V1:/workspace" --workdir /workspace bufbuild/buf:1.8.0 format -d --exit-code
+
   format-csharp:
+    needs: format-protobuf
```

Comment:
This one takes so much time that I prefer having failed rather than waiting for it

Reply:
Need to be changed